jax/_src/pjit.py

```diff
@@ -374,9 +368,48 @@ def cache_miss(*args, **kwargs):
 
   cpp_pjitted_f = wraps(fun)(cpp_pjit_f)
   cpp_pjitted_f._fun = fun
-  type(cpp_pjitted_f).clear_cache = _cpp_pjit_evict_fn
+  cpp_pjitted_f._jit_info = jit_info
+  # TODO(necula): move these to top-level; we don't need to do this for
+  # every jit
+  cpp_jitted_f_class = type(cpp_pjitted_f)
```

Comment:
Why do we need this class thing? I see this is how it was before but I am curious to know the reason :)

Reply:
I attempted to explain this in the PR description. I'll change the description to be cleared.